src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -22,18 +22,57 @@ public InstalledAndTransitivePackageFeed(IEnumerable<PackageCollectionItem> inst
             _transitivePackages = transitivePackages ?? throw new ArgumentNullException(nameof(transitivePackages));
         }
 
+        internal static PackageCollectionItem SelectTransitiveLatestPackage(IGrouping<string, PackageCollectionItem> groupedPackagesById, IEnumerable<PackageCollectionItem> installedPkgs)
+        {
+            if (groupedPackagesById == null)
+            {
+                throw new ArgumentNullException(nameof(groupedPackagesById));
+            }
+
+            if (installedPkgs == null)
+            {
+                throw new ArgumentNullException(nameof(installedPkgs));
+            }
+
+            IEnumerable<PackageCollectionItem> matchedOrigins = groupedPackagesById
+                .Where(x => x.PackageReferences
+                    .Cast<ITransitivePackageReferenceContextInfo>()
```

Comment:
What happens if x.PackageReferences are not of type ITransitivePackageReferenceContextInfo? This will throw and there's no code that safely enforces that groupedPackagesById's PackageReferences are ITransitivePackageReferenceContextInfo's unless I'm missing it, but this method could accept a collection where this is not the case.

Reply:
Added conditional to ensure all inner PackageReference elements are indeed ITransitivePackageReferenceContextInfo based objects